src/base/hooks/BaseTokenWrapperHook.sol

```diff
@@ -154,17 +158,29 @@ abstract contract BaseTokenWrapperHook is BaseHook, DeltaResolver {
 
     /// @notice Deposits underlying tokens to receive wrapper tokens
     /// @param underlyingAmount The amount of underlying tokens to deposit
+    /// @return actualUnderlyingAmount the actual number of underlying tokens used, i.e. to account for rebasing rounding errors
     /// @return wrappedAmount The amount of wrapper tokens received
-    /// @dev Implementing contracts should handle the wrapping operation
-    ///      The base contract will handle settling tokens with the pool manager
-    function _deposit(uint256 underlyingAmount) internal virtual returns (uint256 wrappedAmount);
+    /// @dev Implementing contracts should handle:
+    //    - taking tokens from PoolManager
+    //    - performing the wrapping operation
+    //    - settling tokens on PoolManager
+    function _deposit(uint256 underlyingAmount)
+        internal
+        virtual
+        returns (uint256 actualUnderlyingAmount, uint256 wrappedAmount);
```

Comment:
when `actualUnderlyingAmount` is different to `underlyingAmount` (or the analogous for withdraw), wouldnt this mean that some dust delta might be left open in that currency?
i.e. someone settles 1 ETH, the hook tries to wrap that 1 ETH but only wraps 0.999999 ETH, and then a delta of 0.000001 ETH is left open in ETH?

Reply:
in this case the extra 1 wei is entirely lost due to rounding. if the wei is still held (ie checked via balanceof) then the correct value here is 1 eth